src/bootstrap/compile.rs

```diff
@@ -1136,14 +1136,15 @@ impl Step for Assemble {
             // for `-Z gcc-ld=lld`
             let gcc_ld_dir = libdir_bin.join("gcc-ld");
             t!(fs::create_dir(&gcc_ld_dir));
-            builder.copy(
-                &lld_install.join("bin").join(&src_exe),
-                &gcc_ld_dir.join(exe("ld", target_compiler.host)),
-            );
-            builder.copy(
-                &lld_install.join("bin").join(&src_exe),
-                &gcc_ld_dir.join(exe("ld64", target_compiler.host)),
-            );
+            let bootstrap_compiler = builder.compiler(0, builder.config.build);
+            for flavor in ["ld", "ld64"] {
+                let lld_wrapper_exe = builder.ensure(crate::tool::LldWrapper {
+                    compiler: bootstrap_compiler,
+                    target: target_compiler.host,
+                    flavor_feature: flavor,
```

Comment:
I don't think this will work when doing a cross-compilation.  The stage0 compiler does not have a `std` for the target platform.  

I think fixing this will be quite tricky, since you can't build tools here because that would result in a cyclic dependency (assemble > tool > assemble).   I'm wondering if maybe building the lld wrapper could be pushed to dist::Rustc?  IIRC, lld isn't actually used for building, it is only generated as an artifact for dist.

Reply:
> IIRC, lld isn't actually used for building, it is only generated as an artifact for dist.

It makes sense to me to change that in the future to allow for faster rustc compilation. Maybe push it to `build::Rustc` or even add it as extra binaries produced by the `rustc-binary` crate next to `rustc`?

Reply:
As far as I can see the target std is already built for the stage1 (cross) compiler at this point so using the (target stage - 1) `build_compiler` there should just work(tm).

Reply:
👍  Yea, that should work.  It looks like it builds an extra copy for the *build* compiler, which isn't strictly necessary, but it only takes about a second to build and shouldn't be a problem.